vortex-file/src/open.rs

```diff
@@ -98,105 +96,89 @@ impl VortexOpenOptions {
         Ok(self)
     }
 
-    /// Configure how to split the file into batches for reading.
-    ///
-    /// Defaults to [`SplitBy::Layout`].
-    pub fn with_split_by(mut self, split_by: SplitBy) -> Self {
-        self.split_by = split_by;
-        self
-    }
-
     /// Configure a custom [`SegmentCache`].
     pub fn with_segment_cache(mut self, segment_cache: Arc<dyn SegmentCache>) -> Self {
-        self.segment_cache = Some(segment_cache);
+        self.segment_cache = segment_cache;
         self
     }
 
     /// Disable segment caching entirely.
     pub fn without_segment_cache(self) -> Self {
         self.with_segment_cache(Arc::new(NoOpSegmentCache))
     }
+}
 
-    /// Configure the execution mode
-    pub fn with_execution_mode(mut self, execution_mode: ExecutionMode) -> Self {
-        self.execution_mode = Some(execution_mode);
-        self
+impl VortexOpenOptions<InMemoryVortexFile> {
+    /// Open an in-memory file contained in the provided buffer.
+    pub fn in_memory<B: Into<ByteBuffer>>(buffer: B) -> Self {
+        Self {
+            read: buffer.into(),
+            options: (),
+            ctx: ALL_ENCODINGS_CONTEXT.clone(),
+            layout_ctx: Arc::new(Default::default()),
+            file_size: None,
+            file_layout: None,
+            segment_cache: Arc::new(NoOpSegmentCache),
+            initial_read_size: 0,
+        }
     }
+}
 
-    /// Configure the number of concurrent I/O requests.
-    pub fn with_io_concurrency(mut self, io_concurrency: usize) -> Self {
-        self.io_concurrency = io_concurrency;
-        self
-    }
+impl<R: VortexReadAt> VortexOpenOptions<GenericVortexFile<R>> {
+    const INITIAL_READ_SIZE: u64 = 1 << 20; // 1 MB
 
-    /// Override the default split-by concurrency.
-    ///
-    /// It is recommended to use more split-by concurrency than I/O concurrency to ensure there
-    /// are always I/O operations enqueued.
-    pub fn with_exec_concurrency(mut self, exec_concurrency: usize) -> Self {
-        self.exec_concurrency = Some(exec_concurrency);
-        self
+    pub fn file(read: R) -> Self {
+        Self {
+            read,
+            // TODO(ngates): move this context into the vortex-file crate
+            options: Default::default(),
+            ctx: ALL_ENCODINGS_CONTEXT.clone(),
+            layout_ctx: LayoutContextRef::default(),
+            file_size: None,
+            file_layout: None,
+            segment_cache: Arc::new(InMemorySegmentCache::new(
+                // For now, use a fixed 1GB overhead.
+                CacheBuilder::new(1 << 30),
+            )),
+            initial_read_size: Self::INITIAL_READ_SIZE,
```

Comment:
do we even need this field? seems like it can be part of the `VortexFileOpener` trait, seems like we only use it in that context.